snuba/web/rpc/v1/resolvers/common/trace_item_table.py

```diff
@@ -1,41 +1,113 @@
-        elif column.HasField("formula"):
+def _add_converter(
+    column: Column, converters: Dict[str, Callable[[Any], AttributeValue]]
+) -> None:
+    if column.HasField("key"):
+        if column.key.type == AttributeKey.TYPE_BOOLEAN:
+            converters[column.label] = lambda x: AttributeValue(val_bool=bool(x))
+        elif column.key.type == AttributeKey.TYPE_STRING:
+            converters[column.label] = lambda x: AttributeValue(val_str=str(x))
+        elif column.key.type == AttributeKey.TYPE_INT:
+            converters[column.label] = lambda x: AttributeValue(val_int=int(x))
+        elif column.key.type == AttributeKey.TYPE_FLOAT:
+            converters[column.label] = lambda x: AttributeValue(val_float=float(x))
+        elif column.key.type == AttributeKey.TYPE_DOUBLE:
             converters[column.label] = lambda x: AttributeValue(val_double=float(x))
         else:
             raise BadSnubaRPCRequestException(
-                "column is not one of: attribute, (conditional) aggregation, or formula"
+                f"unknown attribute type: {AttributeKey.Type.Name(column.key.type)}"
             )
+    elif column.HasField("conditional_aggregation"):
+        converters[column.label] = lambda x: AttributeValue(val_double=float(x))
+    elif column.HasField("formula"):
+        converters[column.label] = lambda x: AttributeValue(val_double=float(x))
+        if get_int_config(
+            "enable_formula_reliability", ENABLE_FORMULA_RELIABILITY_DEFAULT
+        ):
+            _add_converter(column.formula.left, converters)
+            _add_converter(column.formula.right, converters)
+    elif column.HasField("literal"):
+        converters[column.label] = lambda x: AttributeValue(val_double=float(x))
+    else:
+        raise BadSnubaRPCRequestException(
+            "column is not one of: attribute, (conditional) aggregation, or formula"
+        )
+
+
+def get_converters_for_columns(
+    columns: Iterable[Column],
+) -> Dict[str, Callable[[Any], AttributeValue]]:
+    """
+    Returns a dictionary of column labels to their corresponding converters.
+    Converters are functions that convert a value returned by a clickhouse query to an AttributeValue.
+    """
+    converters: Dict[str, Callable[[Any], AttributeValue]] = {}
+    for column in columns:
+        _add_converter(column, converters)
+    return converters
+
+
+def _is_sub_column(result_column_name: str, column: Column) -> bool:
+    """
+    returns true if result_column_name is a sub column of column. false otherwise.
+    """
+    # this logic could theoretically cause issue if the user passes in such a column label to a non-subcolumn.
+    # for now, we assume that the user will not do this.
+    return bool(
+        re.fullmatch(rf"{re.escape(column.label)}(\.left|\.right)+", result_column_name)
+    )
+
+
+def _get_reliabilities_for_formula(
+    column: Column, res: Dict[str, TraceItemColumnValues]
+) -> list[Reliability.ValueType]:
```

Comment:
I don't understand why this function doesn't just return the reliability of the column, why does it need to return a list?